common/shared.c

```diff
@@ -2238,16 +2206,22 @@ void ft_parsecsopts(int op, char *optarg, struct ft_opts *opts)
 	}
 }
 
-int ft_parse_rma_opts(int op, char *optarg, struct ft_opts *opts)
+int ft_parse_rma_opts(int op, char *optarg, struct fi_info *hints,
+		      struct ft_opts *opts)
 {
 	switch (op) {
 	case 'o':
 		if (!strcmp(optarg, "read")) {
+			hints->caps |= FI_READ | FI_REMOTE_READ;
 			opts->rma_op = FT_RMA_READ;
 		} else if (!strcmp(optarg, "writedata")) {
+			hints->caps |= FI_WRITE | FI_REMOTE_WRITE;
+			hints->mode |= FI_RX_CQ_DATA;
```

Comment:
This mode bit needs to be set for just "write" as well. Providers (like verbs) would check this in fi_getinfo. The app may or may not end up using RMA writedata functions. The previous code in the tests were also not correct.

Reply:
You lost me here.  Note that the tests already set FI_RMA.  We can duplicate setting FI_RMA here, though it would limit the use case if FI_ATOMIC were desired instead.

We don't know at this point in the code if we're only initiating reads or write, versus being both an initiator and target.